src/screens/OtherProfile/index.js

```diff
@@ -134,14 +157,91 @@ const OtherProfile = () => {
   }, []);
 
   React.useEffect(() => {
+    netInfo.addEventListener((state) => {
+      if (state.isConnected) {
+        initData();
+      } else {
+        handleOfflineMode();
+      }
+    });
     initData();
   }, []);
 
+  React.useEffect(() => {
+    getOtherFeeds();
+  }, [other_id]);
+
   const initData = () => {
+    setInitLoading(true);
     setUserId(params.data.user_id);
     setUsername(params.data.username);
+    fetchOtherProfile();
+  };
+
+  const checkUserBlockHandle = async (userId, callback) => {
+    const status = await netInfo.fetch();
+    if (status.isConnected) {
+      try {
+        const sendData = {
+          user_id: userId
+        };
+        const processGetBlock = await checkUserBlock(sendData);
+        if (callback) callback();
+        if (processGetBlock.status === 200) {
+          setBlockStatus(processGetBlock.data.data);
+        }
+      } catch (e) {
+        if (__DEV__) {
+          console.log(e, 'error');
+        }
+      }
+    }
   };
 
+  const handleSaveDataOtherProfile = (data) => {
+    setDataMain(data);
+    setDataMainBio(data.bio);
+    checkUserBlockHandle(data.user_id);
+    setOtherId(data.user_id);
+    setLoading(false);
+    setInitLoading(false);
+  };
+
+  const fetchOtherProfile = async () => {
+    const status = await netInfo.fetch();
+    if (status.isConnected) {
+      try {
+        handleOfflineMode();
+        const result = await getOtherProfile(params?.data?.username);
+        if (result.code === 200) {
+          handleSaveDataOtherProfile(result.data);
+        }
+        StorageUtils.otherProfileData.setForKey(
+          params?.data?.username,
+          JSON.stringify(result.data)
+        );
+      } catch (e) {
+        if (e.response && e.response.data && e.response.data.message) {
+          SimpleToast.show(e.response.data.message, SimpleToast.SHORT);
+        }
+        setLoading(false);
+        setInitLoading(false);
+      }
+    } else {
+      handleOfflineMode();
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Several new functions have been introduced including `initData`, `checkUserBlockHandle`, `handleSaveDataOtherProfile`, `fetchOtherProfile`, and `handleOfflineMode`. Ensure that these functions are correctly implemented and handle errors properly. Also, make sure that the async operations in these functions do not lead to any race conditions.
```diff
+ const initData = () => {
+   setInitLoading(true);
+   setUserId(params.data.user_id);
+   setUsername(params.data.username);
+   fetchOtherProfile();
+ };
+ const checkUserBlockHandle = async (userId, callback) => {
+   const status = await netInfo.fetch();
+   if (status.isConnected) {
+     try {
+       const sendData = {
+         user_id: userId
+       };
+       const processGetBlock = await checkUserBlock(sendData);
+       if (callback) callback();
+       if (processGetBlock.status === 200) {
+         setBlockStatus(processGetBlock.data.data);
+       }
+     } catch (e) {
+       if (__DEV__) {
+         console.log(e, 'error');
+       }
+     }
+   }
+ };
+ const handleSaveDataOtherProfile = (data) => {
+   setDataMain(data);
+   setDataMainBio(data.bio);
+   checkUserBlockHandle(data.user_id);
+   setOtherId(data.user_id);
+   setLoading(false);
+   setInitLoading(false);
+ };
+ const fetchOtherProfile = async () => {
+   const status = await netInfo.fetch();
+   if (status.isConnected) {
+     try {
+       handleOfflineMode();
+       const result = await getOtherProfile(params?.data?.username);
+       if (result.code === 200) {
+         handleSaveDataOtherProfile(result.data);
+       }
+       StorageUtils.otherProfileData.setForKey(
+         params?.data?.username,
+         JSON.stringify(result.data)
+       );
+     } catch (e) {
+       if (e.response && e.response.data && e.response.data.message) {
+         SimpleToast.show(e.response.data.message, SimpleToast.SHORT);
+       }
+       setLoading(false);
+       setInitLoading(false);
+     }
+   } else {
+     handleOfflineMode();
+   }
+ };
+ const handleOfflineMode = async () => {
+   const cache = await StorageUtils.otherProfileData.getForKey(params?.data?.username);
+   if (cache) {
+     const data = JSON.parse(cache);
+     handleSaveDataOtherProfile(data);
+   } else {
+     setLoading(false);
+     setInitLoading(false);
+   }
+ };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->